tests/test_interface.py

```diff
@@ -0,0 +1,138 @@
+# Test cases can be run with either of the following:
+# python -m unittest discover
+# nosetests -v --rednose --nologcapture
+
+import unittest, mock
+from app import payments
+from app.db import app_db
+from app.db.models import Payment, Detail
+from app.db.interface import PaymentService
+from app.error_handlers import DataValidationError
+
+CC_DETAIL = {'user_name' : 'Jimmy Jones', 'card_number' : '1111222233334444',
+             'expires' : '01/2019', 'card_type' : 'Mastercard'}
+
+DC_DETAIL = {'user_name' : 'John Jameson', 'card_number' : '4444333322221111',
+             'expires' : '02/2020', 'card_type' : 'Visa'}
+
+PP_DETAIL = {'user_name' : 'John Jameson', 'user_email' : 'jj@aol.com'}
+
+CREDIT = {'nickname' : 'my credit', 'user_id' : 1, 'payment_type' : 'credit', 
+          'details' : CC_DETAIL}
+
+DEBIT = {'nickname' : 'my debit', 'user_id' : 2, 'payment_type' : 'debit', 
+         'details' : DC_DETAIL}
+
+PAYPAL = {'nickname' : 'my paypal', 'user_id' : 2, 'payment_type' : 'paypal', 
+          'details' : PP_DETAIL}
+
+#note 'nickname' is spelled wrong
+BAD_DATA = {'nicknam3' : 'my paypal', 'user_id' : 2, 'payment_type' : 'paypal', 
+            'details' : PP_DETAIL}
+
+PP_RETURN = dict(PAYPAL)
+PP_RETURN ['is_default'] = False
+PP_RETURN ['charge_history'] = 0.0
+PP_RETURN ['payment_id'] = 1
+PP_RETURN ['details']['is_linked'] = True
+
+CC_RETURN = dict(CREDIT)
+CC_RETURN['is_default'] = False
+CC_RETURN['charge_history'] = 0.0
+CC_RETURN['payment_id'] = 1
+
+#this should always be added second
+DC_RETURN = dict(DEBIT)
+DC_RETURN['is_default'] = False
+DC_RETURN['charge_history'] = 0.0
+DC_RETURN['payment_id'] = 2
+
+ps = PaymentService()
```

Comment:
Talked with @nvme5 about this and we think that it might be best to place this guy in the `TestInterface` class, either as a class variable or as an instance variable attached to `self`.